src/apps/components/AppFrame/useUpdateAppToken.ts

```diff
@@ -0,0 +1,56 @@
+import { createAppsDebug } from "@dashboard/apps/apps-debug";
+import { DashboardEventFactory, Events } from "@saleor/app-sdk/app-bridge";
+import { useEffect, useRef } from "react";
+
+/**
+ * https://usehooks.com/usePrevious/
+ */
+function usePreviousValue(value) {
+  const ref = useRef();
+
+  useEffect(() => {
+    ref.current = value;
+  }, [value]);
+
+  return ref.current;
+}
```

Comment:
suggestion: it's quite general hook why not put it inside hooks dir

Reply:
depends on your strategy, IMO its better to keep things private until you need them in the other places. Then, you make it generic. 

Its your codebase so you can make a call